src/main/java/org/folio/event/service/AuditOutboxService.java

```diff
@@ -81,77 +82,97 @@ public Future<Integer> processOutboxEventLogs(Map<String, String> okapiHeaders)
-        log.trace("saveOrderLineOutboxLog, po line id={}", poLine.getId());
-        String orderLine = Json.encode(poLine);
-        return saveOutboxLog(conn, action.value(), EntityType.ORDER_LINE, orderLine, okapiHeaders)
-          .onSuccess(reply -> log.info("Outbox log has been saved for order line id: {}", poLine.getId()))
-          .onFailure(e -> log.warn("Could not save outbox audit log for order line with id {}", poLine.getId(), e));
-      })
+      .map(poLine -> saveOutboxLog(conn, okapiHeaders, action.value(), EntityType.ORDER_LINE, poLine.getId(), poLine))
       .toList();
 
     return GenericCompositeFuture.join(futures)
       .map(res -> true)
       .otherwise(t -> false);
   }
 
-  private List<Future<Boolean>> getKafkaFutures(List<OutboxEventLog> eventLogs, Map<String, String> okapiHeaders) {
-    List<Future<Boolean>> futures = new ArrayList<>();
-    for (OutboxEventLog eventLog : eventLogs) {
-      if (EntityType.ORDER == eventLog.getEntityType()) {
-        PurchaseOrder purchaseOrder = Json.decodeValue(eventLog.getPayload(), PurchaseOrder.class);
-        OrderAuditEvent.Action orderAction = OrderAuditEvent.Action.fromValue(eventLog.getAction());
-        futures.add(producer.sendOrderEvent(purchaseOrder, orderAction, okapiHeaders));
-      } else if (EntityType.ORDER_LINE == eventLog.getEntityType()) {
-        PoLine poLine = Json.decodeValue(eventLog.getPayload(), PoLine.class);
-        OrderLineAuditEvent.Action orderLineAction = OrderLineAuditEvent.Action.fromValue(eventLog.getAction());
-        futures.add(producer.sendOrderLineEvent(poLine, orderLineAction, okapiHeaders));
-      }
-    }
-    return futures;
+  /**
+   * Saves piece outbox log.
+   *
+   * @param conn         connection in transaction
+   * @param piece        the audited piece
+   * @param action       action for piece
+   * @param okapiHeaders the okapi headers
+   * @return future with saved outbox log in the same transaction
+   */
+  public Future<Boolean> savePieceOutboxLog(Conn conn,
+                                            Piece piece,
+                                            PieceAuditEvent.Action action,
+                                            Map<String, String> okapiHeaders) {
+    return saveOutboxLog(conn, okapiHeaders, action.value(), EntityType.PIECE, piece.getId(), piece);
   }
 
   private Future<Boolean> saveOutboxLog(Conn conn,
+                                        Map<String, String> okapiHeaders,
                                         String action,
                                         EntityType entityType,
-                                        String entity,
-                                        Map<String, String> okapiHeaders) {
+                                        String entityId,
+                                        Object entity) {
+    String logMessagePart = "for " + entityType + " with id: " + entityId;
+    log.trace("saveOutboxLog {}", logMessagePart);
```

Comment:
we use this pattern with adding :: and use debug level for this
log.debug("saveOutboxLog:: for {}  with id: {}", entityType, entityId)